### PR TITLE
8294357: (tz) Update Timezone Data to 2022d

### PR DESCRIPTION
**Pull request by satoyoshiki**
Please review this PR.  The change include some code changes in ZoneInfoFile.java and TestZoneInfo310.java since tzdata2022d breaks TestZoneInfo310.java.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294357](https://bugs.openjdk.org/browse/JDK-8294357): (tz) Update Timezone Data to 2022d

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * [Sean Coffey](https://openjdk.org/census#coffeys) (@coffeys - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10460/head:pull/10460` \
`$ git checkout pull/10460`

Update a local copy of the PR: \
`$ git checkout pull/10460` \
`$ git pull https://git.openjdk.org/jdk pull/10460/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10460`

View PR using the GUI difftool: \
`$ git pr show -t 10460`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10460.diff">https://git.openjdk.org/jdk/pull/10460.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java [hunk 1]
```diff
@@ -585,9 +585,8 @@ private static ZoneInfo getZoneInfo(String zoneId,
                     // "last", it works for now. From tzdata2020d this hacking
                     // will not work for Asia/Gaza and Asia/Hebron which follow
                     // Palestine DST rules.
-                    if (dom < 0 || dom >= 24 &&
-                                   !(zoneId.equals("Asia/Gaza") ||
-                                     zoneId.equals("Asia/Hebron"))) {
+                    // No need of hacking for Plaestine DST rules from tz2022d
```
Comment (naotoj) [inline src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java:588]: Should this comment be removed, or at least modified to be better explained?

File: test/jdk/sun/util/calendar/zi/TestZoneInfo310.java [hunk 2]
```diff
@@ -196,6 +197,8 @@ public static void main(String[] args) throws Throwable {
                 zid.equals("Europe/Dublin") || // uses "Eire" rule
                 zid.equals("Europe/Prague") ||
                 zid.equals("Asia/Tehran") || // last rule mismatch
+                zid.equals("Asia/Gaza") ||
+                zid.equals("Asia/Hebron") ||
```
Comment (naotoj) [inline test/jdk/sun/util/calendar/zi/TestZoneInfo310.java:201]: It'd be helpful to append `// uses "Palestine" rule`.

**Review by naotoj — COMMENTED** [review]
Some minor comments

File: test/jdk/java/util/TimeZone/TimeZoneData/displaynames.txt [hunk 3]
```diff
@@ -163,11 +163,9 @@ Europe/Simferopol MSK
 Europe/Sofia EET EEST
 Europe/Tallinn EET EEST
 Europe/Tirane CET CEST
-Europe/Uzhgorod EET EEST
 Europe/Vienna CET CEST
 Europe/Vilnius EET EEST
 Europe/Warsaw CET CEST
-Europe/Zaporozhye EET EEST
```
Comment (naotoj) [inline test/jdk/java/util/TimeZone/TimeZoneData/displaynames.txt:170]: Do we need to remove those newly linked zone ids from testing?

Comment (satoyoshiki) [reply test/jdk/java/util/TimeZone/TimeZoneData/displaynames.txt:170]: This file is not maintained by hand but generated from java/util/TimeZone/TimeZoneData/tools/share/Makefile.  
These two Ukraine zones are integrated to Europe/Kyiv and removed from tzdata/europe, so the entries disappeared in this file.

**Review by coffeys — APPROVED** [review]
LGTM

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**